Minor improvements and various cleanup

tambry opened the pull request:
Sadly, there's no emoticon for a cabbage on Github. :cry:

Nekotekina commented on rpcs3/Emu/RSX/RSXThread.cpp:
```diff
@@ -344,6 +344,9 @@ void RSXThread::DoCmd(const u32 fcmd, const u32 cmd, const u32 args_addr, const
 		LOG_WARNING(RSX, "TODO: NV4097_SET_TEX_COORD_CONTROL");
+		const u32 a0 = ARGS(0);
+		u8 texMask2D = a0 & 1;
+		u8 texMaskCentroid = (a0 & 1) >> 4;
```
Must be always zero.

tambry replied:
Will be fixed in the next commit.

Nekotekina commented on rpcs3/Emu/SysCalls/FuncList.cpp:
```diff
@@ -2560,297 +2560,47 @@ std::string SysCalls::GetFuncName(const u64 fid)
-	case 0x3db7914d: return "sceNpScoreGetRankingByNpIdAsync";
-	case 0x3f0808aa: return "sceNpBasicSetPresence";
-	case 0x3f195b3a: return "sceNpCommerceGetProductCategoryResult";
+	case 0x816c6a5f: return "_sceNpSysutilClientFree";
```
What are you trying to change in FuncList.cpp? Please, stop it. It's not for it.

tambry replied:
I ordered the most of sceNp* functions by the name, instead of the ID to match the rest of the function listed.

Nekotekina commented on rpcs3/Emu/SysCalls/Modules/cellFont.cpp:
```diff
@@ -566,12 +632,10 @@ s32 cellFontGetCharGlyphMetricsVertical()
 	{
-		// s_fontInternalInstance->m_bInitialized = false;
-		// s_fontInternalInstance->m_bFontGcmInitialized = false;
 		delete s_fontInternalInstance;
```
Need to ban new/delete usage in rpcs3... With std::unique_ptr, using new/delete is pointless.

tambry replied:
I'll probably conver them to std::unique_ptr then.

Nekotekina commented on rpcs3/Emu/SysCalls/Modules/cellNetCtl.h:
```diff
@@ -260,3 +272,35 @@ struct CellNetCtlNatInfo
+		"INFO_UPNP_CONFIG",
+	};
+
+	return names[code - 1];
```
I wonder, what will happen if code is out of range?

tambry replied:
Will fix that.

Nekotekina commented on rpcs3/Emu/SysCalls/Modules/cellSpurs.cpp:
```diff
@@ -191,7 +191,7 @@ s32 cellSpursEventFlagGetTasksetAddress(vm::ptr<CellSpursEventFlag> eventFlag, v
 // SPURS lock free queue functions
 //
-s32 _cellSpursLFQueueInitialize();
+s32 _cellSpursLFQueueInitialize(vm::ptr<CellSpurs> spurs, vm::ptr<CellSpursTaskset> taskset, vm::ptr<CellSpursLFQueue> lfQueue, vm::cptr<void> queueBuffer, u32 queueSize, u32 queueDepth, u32 queueDirection);
```
This function has only 6 arguments. And it seems you looked in a wrong place for its arguments.

tambry replied:
@Nekotekina Spurs functions can be called either from SPU/PPU threads and they have different arguments depending on which. For example: https://github.com/tambry/rpcs3/blob/cabbage/rpcs3/Emu/SysCalls/Modules/cellSpurs.cpp#L2702.

Nekotekina commented on rpcs3/Emu/SysCalls/Modules/cellSpurs.cpp:
```diff
@@ -3272,9 +3272,27 @@ s32 cellSpursEventFlagGetTasksetAddress(vm::ptr<CellSpursEventFlag> eventFlag, v
+		return CELL_SPURS_TASK_ERROR_INVAL;
+	}
+
+	memset(lfQueue.get_ptr(), 0, sizeof(CellSpursLFQueue));
```
And any SPU-related function (SPURS, cellSync, cellSync2, for example) must be emulated in 100%-precise way. For this one, it doesn't emit any errors, it's a wrapper for `cellSyncLFQueueInitialize` which is already implemented in `cellSync.cpp`.

Nekotekina commented on rpcs3/Emu/SysCalls/Modules/cellSpurs.h:
```diff
@@ -641,6 +650,13 @@ struct set_alignment(128) CellSpursEventFlag
+	u32 padding[32];
+};
+
+CHECK_SIZE_ALIGN(CellSpursEventFlag, 128, 128);
```
It's cellSyncLFQueue, you don't use search properly.

Nekotekina commented:
Well,
1) Check travis errors
2) Revert changes in FuncList.cpp, it's hard to verify and this file is not for reading. So it doesn't make any sense to "reorder" something.
3) Remove enum CellSpursLFQueueDirection
4) Restore removed header files `sceNpCommerce2.h`, `sceNpSns.h` and `sceNpTus.h` because it's bad idea to put all the definitions in one big header file.
5) Revert changes in emucore.vcxproj.filters, it's hard to verify and this file is not for reading.

tambry commented:
@Nekotekina 
1) Fixed it.
2) Someone who made it, did order it by how they are ordered in the documentation. Most of FuncList is ordered properly, like I made it there. I added 6 function IDs and their names. See https://github.com/tambry/rpcs3/blob/c2747f9cf275c19e0019b858df970f581372d8d9/rpcs3/Emu/SysCalls/FuncList.cpp vs https://github.com/tambry/rpcs3/blob/26e7961ed79946f149146c2e4be0da472ce30325/rpcs3/Emu/SysCalls/FuncList.cpp. Only 6 lines have been added (sceNpSns*),
3) Removed.
4) Restored.
5) See https://github.com/tambry/rpcs3/blob/26e7961ed79946f149146c2e4be0da472ce30325/rpcs3/emucore.vcxproj.filters vs https://github.com/tambry/rpcs3/commit/26e7961ed79946f149146c2e4be0da472ce30325#diff-49815d299a737573b21d4fbc6aedd36aL1048. 6 lines after the linked line in the second link are only removed. Everything else stays the same. I have before also did changes like this.

I understand that the the line counting method isn't very good for verifying that nothing has changed except the order. Though I see absolutely no reason to remove the changes in emucore.vcxproj.filters and FuncList.cpp. Also in the solution all the given files properly appear, which shows that anything isn't missing from emucore.vcxproj.filters. I can't find any good tools to further prove that nothing in those files has technically changed, except the order.

Nekotekina commented:
You don't need to add lines in FuncList.cpp. Why? Because every function registered by REG_FUNC will be registered with its name and this name will be acquired correctly anyway. You may look at the bottom of FuncList.cpp file to see how it works. But not every function is registered by REG_FUNC so we decided to just not touch this file. It's really hard to verify.

If some function appears without ID, everything you need is just call REG_FUNC properly, nothing more.

tambry commented:
@Nekotekina Then just one question: Why does FuncList even exist then?

Nekotekina commented:
1) It's old. But it's not completely obsolete. I already said the reason: not every function is registered by REG_FUNC currently.
2) It was generated automatically at some point of time. If you do changes, it's hard to verify because who knows whether you added some ID correctly or not.

I suggest you to redo all the commits because unwanted changes appeared and the history is unclean, you may rewrite the history completely, for example, or do in another branch.

tambry commented:
I'll do a new one tomorrow. On a new branch most likely.

tambry commented:
@Nekotekina Okey, this should be fixed up now.

Nekotekina commented:
I'm testing it.
